### PR TITLE
fix(convert-v5): pre-resolve module dependencies in generated mission.yaml

### PR DESCRIPTION
**Pull request by davidp57**
## Problem

`convert-v5` generated `mission.yaml` without resolving module dependencies. A later `veaf-tools build` then had to auto-enable missing dependencies at config-generation time and print a warning, e.g.:

> Module 'CASMISSION' requires 'GROUNDAI' which is not configured — auto-enabling 'GROUNDAI'

Since the converter had just produced the file, it should emit a self-consistent `mission.yaml` in the first place.

## Fix

- **`lua_config_generator.py`**: new pure helper `resolve_module_dependencies(enabled_ids)` that walks the existing `_MODULE_DEPS` graph **transitively** (no logging, no mutation). The converter reuses this single source of truth instead of duplicating the graph.
- **`v5_converter._build_mission_yaml`**: pre-resolves dependencies and emits the required modules explicitly as enabled (e.g. `CASMISSION` → `GROUNDAI`, `SPAWN`, and their transitive deps). The auto-enabled modules are recorded on the `ConversionReport` and listed in the conversion report.

The build-time `_resolve_deps` safety net still runs, but now finds the dependencies already present, so no warning is emitted and the generated file accurately reflects what will run.

## Tests
- `resolve_module_dependencies`: transitive closure, skips already-enabled modules, empty for leaf/empty input.
- `_build_mission_yaml`: enabling `CASMISSION` yields `GROUNDAI` + `SPAWN` in the YAML and records them; no deps recorded when none needed; the report mentions the resolved modules.
- Full suite green; `ruff check` / `ruff format --check` / `mypy` clean on `src/python`.

## Backlog
- Closes lot `FIX-CONVERT-V5-DEPS` (tickets `CVDEP-001..003`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Pre-resolve module dependencies when generating mission.yaml so convert-v5 outputs a self-consistent configuration and surfaces auto-enabled modules in the conversion report.

Bug Fixes:
- Ensure convert-v5-generated mission.yaml auto-enables required module dependencies (e.g. CASMISSION → GROUNDAI, SPAWN, and transitive deps) so builds no longer emit dependency auto-enable warnings.

Enhancements:
- Introduce a pure resolve_module_dependencies helper in lua_config_generator to expose transitive dependency resolution without side effects.
- Extend ConversionReport to record and display auto-resolved module dependencies in the mission.yaml section of the report.

Tests:
- Add unit tests for resolve_module_dependencies covering transitive closure, already-enabled modules, and leaf/empty inputs.
- Add integration-style tests for V5Converter._build_mission_yaml to verify dependency auto-enabling, reporting, and absence of deps when not needed.

Chores:
- Mark FIX-CONVERT-V5-DEPS backlog lot and related CVDEP tickets as completed, bump project version to 6.3.12, and update the changelog with the new convert-v5 behavior.

### CODE REVIEW

File: src/python/veaf-tools/mission_builder/v5_converter.py [hunk 1]
```diff
@@ -919,6 +929,13 @@ def _build_mission_yaml(self, report: ConversionReport) -> str:
 
         enabled_modules = mr.enabled_modules if mr else []
         enabled_set = set(enabled_modules) | _BASE_ALWAYS_ON
+        # Pre-resolve module dependencies (e.g. CASMISSION → GROUNDAI, SPAWN) so
+        # the generated mission.yaml is self-consistent and the build no longer
+        # needs to auto-enable them at config-generation time with a warning.
+        auto_deps = resolve_module_dependencies(enabled_set)
+        if auto_deps:
+            enabled_set.update(auto_deps)
+            report.auto_resolved_deps = auto_deps
```
[the remark this answers is not in the document]
Comment (davidp57) [reply src/python/veaf-tools/mission_builder/v5_converter.py]: Fixed in f41964a — `report.auto_resolved_deps` is now assigned unconditionally (empty list when there are no auto-deps), so a reused report never keeps stale values.
